### DISCUSSION TITLE
[Markdown] Respect all whitespace characters

### DISCUSSION THREAD

**Discussion by James-E-A** [community/community#110515] [Projects and Issues]
### Select Topic Area

Product Feedback

### Feature Area

Issues

### Body

I wish that the Markdown parser you use had better whitespace detection, particularly when it comes to auto-linkifying commits.

## Current Behavior (undesirable)

For example, if I type the following fragment:

```markdown
Another alternative fix would be to revert the /IMPLIB injection code to its pre-06830ada1cda3595d2a1f222fd8cbed35bdb691e behavior, just nestling it in a top-level …
```

It renders as:

> Another alternative fix would be to revert the /IMPLIB injection code to its pre-06830ada1cda3595d2a1f222fd8cbed35bdb691e behavior, just nestling it in a top-level …

This is a bummer because it doesn't link the issue, which is a major lack of functionality.

### Current workaround #1 (undesirable)

If I stick a space after &ldquo;pre-&rdquo;, it works:

```markdown
Another alternative fix would be to revert the /IMPLIB injection code to its pre- 06830ada1cda3595d2a1f222fd8cbed35bdb691e behavior, just nestling it in a top-level …
```

> Another alternative fix would be to revert the /IMPLIB injection code to its pre- <a href="https://github.com/pypa/setuptools/commit/06830ada1cda3595d2a1f222fd8cbed35bdb691e"><code>06830ad</code></a> behavior, just nestling it in a top-level …

But this is a bummer because it's pretty hideous.

### Current workaround #2 (undesirable)

If I _manually_ format it, it works:

```markdown
Another alternative fix would be to revert the /IMPLIB injection code to its pre-[`06830ad`](https://github.com/pypa/setuptools/commit/06830ada1cda3595d2a1f222fd8cbed35bdb691e) behavior, just nestling it in a top-level …
```

> Another alternative fix would be to revert the /IMPLIB injection code to its pre-<a href="https://github.com/pypa/setuptools/commit/06830ada1cda3595d2a1f222fd8cbed35bdb691e"><code>06830ad</code></a> behavior, just nestling it in a top-level …

This is a bummer because it's extra work.

## Requested workaround

The Zero-Width Non-Joiner character was **made for** this use-case, where you want two semantically- and whitespace-separated "words" to be displayed physically "without" any whitespace between them, even though the whitespace *logically* exists:

```markdown
Another alternative fix would be to revert the /IMPLIB injection code to its pre-&#x200c;06830ada1cda3595d2a1f222fd8cbed35bdb691e behavior, just nestling it in a top-level …
```

but Github doesn't currently respect this, rendering the HTML *exactly* as if there were no whitespace characters surrounding the commit hash.

\(I seem to be [the first](https://github.com/orgs/community/discussions?discussions_q=Zero-Width+Non-Joiner) to ask this.\)

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @JamesTheAwesomeDude ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Projects and Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. Good luck!

**Reply by James-E-A** [reply]
I see... I thought that since my request was about how *Issues* (and comments on *Issues*) are formatted, it belonged in that forum.

**Reply by queenofcorgis** [reply]
@JamesTheAwesomeDude thanks for clarifying and I apologize I saw commits and my mind went elsewhere. I've moved it back and added the appropriate tags. Have a great weekend!
